detection.py

```diff
@@ -1,289 +1,133 @@
-
-    emd, s_sim = None, None
-    try:
-        emd = cl.earth_movers_distance(path_a, path_b)
-    except Exception:
-        main_logger.exception('Error calculating earth_movers_distance')
-
-    try:
-        s_sim = cl.structural_sim(path_a, path_b)
-    except Exception:
-        main_logger.exception('Error calculating structural_sim')
-
-    main_logger.info(f"Compared url '{resulturl}'")
-    main_logger.info(f"Finished comparing:  emd = '{emd}', structural_sim = '{s_sim}'")
-
-    if ((emd < 0.001) and (s_sim > 0.70)) or ((emd < 0.002) and (s_sim > 0.80)):
-        return True
-
-    return False
-
-
-async def check_search_results(url_registered_domain, found_urls) -> "DetectionResult":
-    with TimeIt("SAN domain check"):
-        with concurrent.futures.ThreadPoolExecutor() as pool:
-            loop = asyncio.get_running_loop()
-            coros = []
-            for url in found_urls:
-                coros.append(loop.run_in_executor(pool, lambda: check_url(url_registered_domain, url)))
+    for method in settings.detection_methods:
+        main_logger.info(f"Started running method {method.name}")
+        results.append(method.value.test(data.url, data.screenshot_url, data.uuid, data.pagetitle, None))
 
-            for coro in asyncio.as_completed(coros):
-                if await coro:
-                    return True
+    result = decide(settings.decision_strategy, results)
+    session.set_state(result.name, "DONE")
 
-        # If no match, no results yet
-        return False
-    
-def check_url(url_registered_domain, url) -> "DetectionResult":
-    # For each found URL, get the hostname
-    domain = domains.get_hostname(url)
+    return DetectionResult(data.url, url_hash, "DONE", result)
 
-    # Get the Subject Alternative Names (all associated domains, e.g. google.com, google.nl, google.de) for all websites
-    try:
-        san_names = domains.get_san_names(domain)
-    except Exception:
-        main_logger.error(f"Error in SAN for {domain} (from URL {url})", exc_info=1)
-        return
 
-    main_logger.debug(f"Domain of URL `{url}` is {domain}, with SAN names {san_names}")
+class DetectionSettings:
+    detection_methods: list[DetectionMethods]
+    decision_strategy: DecisionStrategies
+    bypass_cache: bool = False
 
-    for hostname in [domain] + san_names:
-        # Check if any of the domains found matches the input domain
-        registered_domain = domains.get_registered_domain(hostname)
-        if url_registered_domain == registered_domain:
-            return True
-    return False
+    def __init__(
+        self,
+        detection_methods: list[DetectionMethods] = [DetectionMethods.ReverseImageSearch],
+        decision_strategy: DecisionStrategies = DecisionStrategies.Majority,
+        bypass_cache: bool = False,
+    ):
+        self.detection_methods = detection_methods
+        self.decision_strategy = decision_strategy
+        self.bypass_cache = bypass_cache
 
+    @classmethod
```

Comment:
wouldnt it be better to make this static instead? otherwise calling it with an instance will overwrite its data

Reply:
same for DetectionData